cachi2/core/package_managers/gomod.py

```diff
@@ -0,0 +1,1022 @@
+        if should_vendor:
+            module_lines = _module_lines_from_modules_txt(request.source_dir)
+        else:
+            # .String formats the module as <name> <version> [=> <replace>],
+            #   where <replace> is <name> <version> or <path>
+            output_format = "{{ if not .Main }}{{ .String }}{{ end }}"
+            go_list_output = _run_gomod_cmd(
+                ("go", "list", "-mod", "readonly", "-m", "-f", output_format, "all"), run_params
+            )
+            module_lines = go_list_output.splitlines()
+
+        module_level_deps = []
+        # Keep track of which dependency replacements were actually applied to verify they were all
+        # used later
+        used_replaced_dep_names = set()
+        for line in module_lines:
+            parts = line.split(" ")
+
+            replaces = None
+            if len(parts) == 4 and parts[2] == "=>":
+                # If a Go module uses a "replace" directive to a local path, it will be shown as:
+                # k8s.io/metrics v0.0.0 => ./staging/src/k8s.io/metrics
+                # In this case, take the module name and the relative path, since that is the
+                # actual dependency being used.
+                parts = [parts[0], parts[-1]]
+            elif len(parts) == 5 and parts[2] == "=>":
+                # If a Go module uses a "replace" directive, then it will be in the format:
+                # github.com/pkg/errors v0.8.0 => github.com/pkg/errors v0.8.1
+                # In this case, just take the right side since that is the actual
+                # dependency being used
+                old_name, old_version = parts[:2]
+                # Only keep track of user provided replaces. There could be existing "replace"
+                # directives in the go.mod file, but they are an implementation detail specific to
+                # Go and they don't need to be recorded in Cachito.
+                if old_name in replaced_dep_names:
+                    used_replaced_dep_names.add(old_name)
+                    replaces = {"type": "gomod", "name": old_name, "version": old_version}
+                parts = parts[3:]
+
+            if len(parts) == 2:
+                module_level_deps.append(
+                    {"name": parts[0], "replaces": replaces, "type": "gomod", "version": parts[1]}
+                )
+            else:
+                log.warning("Unexpected go module output: %s", line)
+
+        unused_dep_replacements = replaced_dep_names - used_replaced_dep_names
+        if unused_dep_replacements:
+            raise GoModError(
+                "The following gomod dependency replacements don't apply: "
+                f'{", ".join(unused_dep_replacements)}'
+            )
+
+        # In case a submodule is being processed, we need to determine its path
+        subpath = (
+            None
+            if path == git_dir_path
+            else path.replace(f"{git_dir_path}/", "")
```

Comment:
`path` is Path here, hence doesn't have a replace() method
```suggestion
            else path.relative_to(git_dir_path)
```

Reply:
Fixed it;